plugins/c8y_configuration_plugin/src/download.rs

```diff
@@ -309,6 +329,12 @@ impl ConfigDownloadRequest {
         let src = &self.tmp_dir.join(&self.file_name);
         let dest = &self.file_path;
 
+        if let Some(dir_to) = dest.parent() {
+            if !dest.exists() {
+                fs::create_dir_all(dir_to)?;
```

Comment:
Might be better to use tokio to create the directories here?
```suggestion
                tokio::fs::create_dir_all(dir_to).await?;
```

Reply:
Couldn't beacuse this `move_file` function is not async. Changing it to `async` didn't make much sense either as using `fs` was much simpler.

Reply:
But isn't it in an async function now?

Reply:
> But isn't it in an async function now?

Nope. This piece of code is added to the `ConfigDownloadRequest::move_file` function which isn't async currently (none of the other members of this struct are async either). It **can be** made async as there's only one other function that's using it, but I don't really see any benefit in doing it that. Do you see any?

Reply:
>  It can be made async as there's only one other function that's using it, but I don't really see any benefit in doing it that. Do you see any?

As said by [tokio/fs](https://docs.rs/tokio/1.21.2/tokio/fs/index.html) documentation: 

> Tasks run by worker threads should not block, as this could delay servicing reactor events. Portable filesystem operations are blocking, however.

In that specific case, I don't think this is a big deal.